### PR TITLE
docs: Add Contributing document

### PR DESCRIPTION
**Pull request by avaly**
To attract more open source contributors we should provide some basic guidelines of contributing to the repository.

### CODE REVIEW

**Review by ejmartin504 — COMMENTED** [review]
Looks good overall! If this is adapted from somewhere else, we should attribute it, no?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+1. Make your code or other changes.
+1. Add or update relevant unit/type tests.
+1. Push the code to your forked repository and open a draft pull request (please follow the guidelines in [How to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/))
+1. Perform a self-review using the reviewer guidelines below prior to taking the PR out of draft state.
+
+### Reviewer Guidelines
+
+Reviewers should use the following questions to evaluate the implementation for correctness/completeness and ensure all housekeeping items have been addressed prior to merging the code.
+
+- Correctness/completeness
```
Comment (sullman) [inline CONTRIBUTING.md:90]: Is it worth mentioning anything about semver here for thinking about how a change should be released/versioned?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+```
+
+This helps us automate the [HISTORY.md](HISTORY.md) generation.
+
+## Pull Request Process
+
+1. Create your branch from `main`
+1. Make your code or other changes.
+1. Add or update relevant unit/type tests.
+1. Push the code to your forked repository and open a draft pull request (please follow the guidelines in [How to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/))
```
[the remark this answers is not in the document]
Comment (avaly) [reply CONTRIBUTING.md:83]: This whole file is a mixture of the [Babel](https://github.com/babel/babel/blob/main/CONTRIBUTING.md) and [mongodb](https://github.com/mongodb/node-mongodb-native/blob/main/CONTRIBUTING.md)'s contributing files.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+```
+
+### Commit messages
+
+This repository is following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
+
+The format should look something like the following (note the blank lines):
+
+```txt
+<type>[optional scope]: <description>
```
[the remark this answers is not in the document]
Comment (avaly) [reply CONTRIBUTING.md:69]: I got this code from the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) site itself.

[the remark this answers is not in the document]
Comment (ejmartin504) [reply CONTRIBUTING.md:69]: Weird that they don't include the parens 🤷 

Thoughts on adding list of scopes?

[the remark this answers is not in the document]
Comment (avaly) [reply CONTRIBUTING.md:69]: We don't really use scopes, but I did list all the types we support.

File: CONTRIBUTING.md (hunk 1 above)
Comment (avaly) [reply CONTRIBUTING.md:110]: I've added a section about semver in the commit messages above.

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ejmartin504) [reply CONTRIBUTING.md:69]: I've seen "docs" and "release" used FWIW

**Comment by vanstinator** [comment]
Looks good!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+Running tests in watch mode:
+
+```sh
+$ yarn watch
+$ yarn watch:types
+```
+
+### Commit messages
+
+This repository is following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
```
Comment (ejmartin504) [inline CONTRIBUTING.md:64]:
```suggestion
This repository follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
```

**Review by ejmartin504 — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (avaly) [reply CONTRIBUTING.md:69]: `core(release)` is being added by the `standard-version` tool. All other human made commits don't really specify a scope.

[the remark this answers is not in the document]
Comment (ejmartin504) [reply CONTRIBUTING.md:69]: Wow I could have _sworn_ I saw others when I looked at the commit history. Weird. Must have been a dream!

**Pull request merged**